watertap/costing/units/anaerobic_digestor.py

```diff
@@ -0,0 +1,93 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2023, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+import pyomo.environ as pyo
+from ..util import (
+    register_costing_parameter_block,
+    make_capital_cost_var,
+)
+
+
+def build_anaerobic_digestor_cost_param_block(blk):
+
+    # NOTE: costing data are from anaerobic_digestion_reative.yaml
+    blk.capital_a_parameter = pyo.Var(
+        initialize=19.3552312e6,
+        doc="A parameter for capital cost",
+        units=pyo.units.USD_2012,
+    )
+    blk.capital_b_parameter = pyo.Var(
+        initialize=0.6,
+        doc="B parameter for capital cost",
+        units=pyo.units.dimensionless,
+    )
+    blk.reference_flow = pyo.Var(
+        initialize=911.054,
+        doc="Reference flow for capital cost",
+        units=pyo.units.m**3 / pyo.units.hr,
+    )
+
+
+@register_costing_parameter_block(
+    build_rule=build_anaerobic_digestor_cost_param_block,
+    parameter_block_name="anaerobic_digestor",
+)
+def cost_anaerobic_digestor(blk, cost_electricity_flow=True):
+    """
+    Anaerobic digestor costing method
+    """
+    cost_anaerobic_digestor_capital(
+        blk,
+        blk.costing_package.anaerobic_digestor.capital_a_parameter,
+        blk.costing_package.anaerobic_digestor.capital_b_parameter,
+        blk.costing_package.anaerobic_digestor.reference_flow,
+    )
+
+    t0 = blk.flowsheet().time.first()
+    if cost_electricity_flow:
+        blk.costing_package.cost_flow(
+            pyo.units.convert(
+                blk.unit_model.electricity[t0],
+                to_units=pyo.units.kW,
+            ),
+            "electricity",
+        )
+
+
+def cost_anaerobic_digestor_capital(
```

Comment:
We should eventually consider and account for o&m costs, but this is a start.

Reply:
I added the electricity cost in the unit model files.

Reply:
That's good and should be good enough for this PR, but later, we might consider other costs such as heating and other relevant o&m.